app/code/core/Mage/Checkout/controllers/CartController.php

```diff
@@ -711,4 +714,15 @@ protected function _getCustomerSession()
     {
         return Mage::getSingleton('customer/session');
     }
+
+    /**
+     * Set product form data in checkout session for populating the product form
+     * in case of errors in add to cart process.
+     */
+    protected function _setProductBuyRequest()
```

Comment:
```suggestion
    protected function _setProductBuyRequest(): void
```

or this? not sure if it's the right syntax